v-next/hardhat-node-test-reporter/src/github-actions.ts

```diff
@@ -0,0 +1,63 @@
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+
+import { formatError } from "./error-formatting.js";
+import { cleanupTestFailError } from "./node-test-error-utils.js";
+import { TestEventData } from "./node-types.js";
+
+export async function annotatePR(
+  event: TestEventData["test:fail"],
+): Promise<void> {
+  if (
+    process.env.GITHUB_ACTIONS === undefined ||
+    process.env.NO_GITHUB_ACTIONS_ANNOTATIONS !== undefined
+  ) {
+    return;
+  }
+
+  const error = event.details.error;
+
+  const location = getLocation(error);
+  if (location === undefined) {
+    return;
+  }
+
+  const { default: core } = await import("@actions/core");
```

Comment:
do we need to import packages dynamically in the reporter?

Reply:
TBH it's unclear. Not very well documented by github if this package is safe outside of github actions. 

Also, locally it won't be used it, so it makes sense not to import it.